fx quant: add docblocks to _find_matches and _find_quants

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#43928 fx quant: add docblocks to _find_matches and _find_quants**

Summary:

Improving readability, no logic change.

Test Plan:

CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23440249](https://our.internmc.facebook.com/intern/diff/D23440249)

vkuzo committed:
fx quant: add docblocks to _find_matches and _find_quants

Summary:

Improving readability, no logic change.

Test Plan:

CI

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on torch/quantization/fx/quantize.py:
```diff
@@ -955,6 +967,14 @@ def record_match(pattern, node, matched):
 
     def _find_quants(self, graph, matches):
+        """
+        Inputs:
```
would also be good to describe what this is doing.
this function finds the input and output nodes of quantizable module/functional/torch ops that also needs to be quantized and return a map from node name to a tuple of QuantizeHandler and qconfig

jerryzh168 commented on torch/quantization/fx/quantize.py:
```diff
@@ -955,6 +967,14 @@ def record_match(pattern, node, matched):
+          - matches: output of self._find_matches function
+
+        Outputs a map of
+          node_name -> (DefaultQuant instance, qconfig)
```
QuantizeHandler instance would be better I think

vkuzo replied:
what would be the context for that?  Seems like only `DefaultQuant` instances are added in this function

jerryzh168 replied:
DefaultQuant is inheriting from QuantizeHandler, QuantizeHandler occurred in other places as well so it would be clearer to use it here I think. we can roughly think about QuantizeHandler as the type and DefaultQuant as specific instance

jerryzh168 replied:
basically find_matches find QuantizeHandler for quantizable ops and find_quants find QuantizeHandler for inputs/outputs of these ops

jerryzh168 commented on torch/quantization/fx/quantize.py:
```diff
@@ -928,7 +928,19 @@ def get_next_i(module, qparams):
 
     def _find_matches(self, graph, modules, patterns):
-        match_map = {}  # node name -> (root_node, match_value?)
+        """
```
same for this one, a brief description for what this function does would help understanding as well

vkuzo replied:
sure, I was going to add more details in future PRs, but we can add it here

vkuzo committed:
Update on "fx quant: add docblocks to _find_matches and _find_quants"


Summary:

Improving readability, no logic change.

Test Plan:

CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23440249](https://our.internmc.facebook.com/intern/diff/D23440249)

[ghstack-poisoned]

vkuzo committed:
Update on "fx quant: add docblocks to _find_matches and _find_quants"


Summary:

Improving readability, no logic change.

Test Plan:

CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23440249](https://our.internmc.facebook.com/intern/diff/D23440249)

[ghstack-poisoned]

vkuzo committed:
Update on "fx quant: add docblocks to _find_matches and _find_quants"


Summary:

Improving readability, no logic change.

Test Plan:

CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23440249](https://our.internmc.facebook.com/intern/diff/D23440249)

[ghstack-poisoned]

jerryzh168 commented on torch/quantization/fx/quantize.py:
```diff
@@ -529,7 +529,27 @@ def convert(self, observed, inplace=False, debug=False, is_dynamic=False):
+          - patterns: a mapping from a tuple of nodes in reverse order to
+              uninitialized QuantizeHandler subclass.
+
+        Matches the nodes in the input graph to quantization patterns, and
+        outputs the information needed to quantize them in future steps.
```
nit: the order is typically: description, args, returns, example

jerryzh168 reviewed (approved):
please change the order of docs

vkuzo committed:
Update on "fx quant: add docblocks to _find_matches and _find_quants"


Summary:

Improving readability, no logic change.

Test Plan:

CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23440249](https://our.internmc.facebook.com/intern/diff/D23440249)

[ghstack-poisoned]

vkuzo committed:
Update on "fx quant: add docblocks to _find_matches and _find_quants"


Summary:

Improving readability, no logic change.

Test Plan:

CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23440249](https://our.internmc.facebook.com/intern/diff/D23440249)

[ghstack-poisoned]

vkuzo committed:
Update on "fx quant: add docblocks to _find_matches and _find_quants"


Summary:

Improving readability, no logic change.

Test Plan:

CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23440249](https://our.internmc.facebook.com/intern/diff/D23440249)

[ghstack-poisoned]

vkuzo committed:
Update on "fx quant: add docblocks to _find_matches and _find_quants"


Summary:

Improving readability, no logic change.

Test Plan:

CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23440249](https://our.internmc.facebook.com/intern/diff/D23440249)

[ghstack-poisoned]
